torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3297,52 +3297,79 @@ struct to_ir {
     return emitBuiltinCall(loc, *graph, aten::select, {input, dim, index}, {});
   }
 
-  // Desugars slice indexing: tensor[begin:end] -> tensor.slice(dim, begin, end,
-  // 1)
-  Value* emitSlice(
+  Value* emitSliceOp(
       const SourceRange& loc,
-      Value* input,
-      Value* dim, // Only used for tensor slicing
-      const SliceExpr& slice) {
+      Value* sliceable,
+      Value* dim,
+      Value* start,
+      Value* end,
+      Value* step) {
     std::vector<NamedValue> args;
     args.reserve(4);
-    args.emplace_back(loc, "self", input);
+    args.emplace_back(loc, "self", sliceable);
 
     // XXX: If list slicing becomes more complicated or stops using
     // aten::slice, we should separate it from this function.
     if (dim) {
-      AT_ASSERT(input->type()->isSubtypeOf(TensorType::get()));
+      AT_ASSERT(sliceable->type()->isSubtypeOf(TensorType::get()));
 
       args.emplace_back(dim);
     } else {
-      AT_ASSERT(!input->type()->isSubtypeOf(TensorType::get()));
+      AT_ASSERT(!sliceable->type()->isSubtypeOf(TensorType::get()));
+    }
+
+    // Default value for start is 0.
+    if (!start) {
+      start = graph->insertConstant(0, loc);
     }
+    args.emplace_back(loc, "start", start);
 
-    args.emplace_back(loc, "begin", emitExpr(Expr(slice.startOr(0))));
```

Comment:
Note to reviewer:

I changed name of this argument from "begin" to "start" to match the schema of `aten::slice`. I think it worked before because this argument is always used as positional argument. Let me know if there are other things to consider.